### DISCUSSION TITLE
Is it possible to import react component into snabbdom view

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1033] [Q&A] [answered]
Hi,
I create a custom view for my model. I read workflow's Iconview code and create my custom view as follows
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/4718545a-a501-4900-a980-4809a29e8192)
I want to use Theia's built-in ConfirmDialog when I click the button in my custom view, ConfirmDialog is a react dialog.
my problem is that it seems to react component is not compatible with snabbdom view, I get an error "Type 'VNode' is not assignable to type 'ReactNode'", Is it possible to import react component into snabbdom view?
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/e8ccbe1f-35e7-44a6-b29d-0857cba3bbcc)
1. I think it is possible, because when I use yarn watch, and add import { ConfirmDialog, Dialog } from '@theia/core/lib/browser/dialogs'; my custom work correctly I can click the button and show ConfirmDialog.
2. when I exit watch mode, I get the error above and I can not execute the yarn build.
3. I just do not know how to set and make the error disappear, who has experience

**Comment by Adem-Kri** [comment]
I just had a similar issue #1032 and haven't found a solution yet,

As a shortcut, you can add type any to buttonNode, it helped me to bypass the issue and take care of other things
```ts
const buttonNode: any = this.createNode(model.args);
``` 
I hope this helps

**Comment by martin-fleck-at** [comment, marked as answer]
@wss29 That is an interesting problem! My assumption is that it the JSX system gets confused about what to properly emit and wrongly assumes we are targeting React. Unfortunately, I have no idea how to fix that. I believe the easiest workaround would be to simply "cast" everything to make the compiler happy, e.g., something like:

```
        const node = (
            <g>
                <foreignObject requiredFeatures='http://www.w3.org/TR/SVG11/feature#Extensibility' height={50} width={70} x={0} y={0}>
                    {buttonNode as ReactNode}
                </foreignObject>
            </g>
        );
        const vnode = node as unknown as VNode;
```

**Comment by wss29** [comment]
Hi,
    Thanks guys.
